modules/optics/arrow-optics/src/main/kotlin/arrow/optics/Iso.kt

```diff
@@ -69,22 +69,26 @@ interface PIso<S, T, A, B> : PIsoOf<S, T, A, B> {
     /**
      * Lift a [PIso] to a Functor level
      */
-    fun <F> mapping(FF: Functor<F>): PIso<Kind<F, S>, Kind<F, T>, Kind<F, A>, Kind<F, B>> = PIso(
```

Comment:
Unrelated to this PR but upon seeing this I was wondering do we want to completely remove ` fun <F, A, B> Functor<F>.map(fa: Kind<F, A>. f: (A) -> B): Kind<F, B>`? Because in this case or [here](https://github.com/arrow-kt/arrow/pull/745/files#diff-5e328743e478d4b73e209941fff61b3dL86) I am in doubt that it is an improvement.

Maybe it is just something I need to get used to though but you're forced to use `with` or `run` now. Our `Functor` definition also looks really different from what people are used to for `Functor`.

Thoughts?

Reply:
Where is that extension function `Functor<F>.map` defined? AFAIK it was deleted.

>Maybe it is just something I need to get used to though but you're forced to use with or run now.

I need to write a couple of blog posts and docs about the new intuitions for typeclasses. Functor now works in a similar way to how it does in Haskell, where it's a set of functions parametrized to a single type and available at a scope. For people coming from Java we can now say that Typeclasses are like DSLs to create Kinds.